File: daily_question/1544._make_the_string_great/Go/main.go
package main

import "fmt"

func makeGood(s string) string {
	stack := make([]rune, 0)
	for _, char := range s {
		if len(stack) > 0 && abs(int(char)-int(stack[len(stack)-1])) == 32 {
			stack = stack[:len(stack)-1]
			continue
		}
		stack = append(stack, char)
	}

	return string(stack)
}

func abs(x int) int {
	if x < 0 {
		return -x
	}

	return x
}

func main() {
	s := "leEeetcode"
	result := makeGood(s)
	fmt.Println(result == "leetcode")

	s = "abBAcC"
	result = makeGood(s)
	fmt.Println(result == "")

	s = "s"
	result = makeGood(s)
	fmt.Println(result == "s")
}

File: daily_question/1544._make_the_string_great/python3/main.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack: list = []
        for char in s:
            if stack and abs(ord(char) - ord(stack[-1])) == 32:
                stack.pop()
            else:
                stack.append(char)

        return ''.join(stack)


if __name__ == "__main__":
    solution: Solution = Solution()

    assert solution.makeGood("leEeetcode") == "leetcode"
    assert solution.makeGood("abBAcC") == ""
    assert solution.makeGood("s") == "s"
